public/app/features/datasources/settings/DataSourceSettingsPage.tsx

```diff
@@ -234,12 +234,14 @@ export class DataSourceSettingsPage extends PureComponent<Props> {
         )}
 
         <div className="gf-form-group">
-          {testingStatus && testingStatus.message && (
+          {testingStatus?.message && (
             <Alert
               severity={testingStatus.status === 'error' ? 'error' : 'success'}
               title={testingStatus.message}
               aria-label={selectors.pages.DataSource.alert}
-            />
+            >
+              {testingStatus.details?.message ?? null}
```

Comment:
IIUC, this means if the details returns a `message` key, it will be displayed.  Can we add that as a comment on the `Record<string, any>` details constraint?